File: Schedule/MainForm.rb
require "mscorlib"
require "System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

class MainForm < System::Windows::Forms::Form
	def initialize()
		self.InitializeComponent()
	end

	def InitializeComponent()
		@label1 = System::Windows::Forms::Label.new()
		@label2 = System::Windows::Forms::Label.new()
		@label3 = System::Windows::Forms::Label.new()
		@label4 = System::Windows::Forms::Label.new()
		@label5 = System::Windows::Forms::Label.new()
		@label6 = System::Windows::Forms::Label.new()
		@label7 = System::Windows::Forms::Label.new()
		@label8 = System::Windows::Forms::Label.new()
		@button1 = System::Windows::Forms::Button.new()
		@button2 = System::Windows::Forms::Button.new()
		self.SuspendLayout()
		# 
		# label1
		# 
		@label1.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label1.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label1.Location = System::Drawing::Point.new(88, 45)
		@label1.Name = "label1"
		@label1.Size = System::Drawing::Size.new(105, 42)
		@label1.TabIndex = 0
		@label1.Click { |sender, e| self.Label1Click(sender, e) }
		# 
		# label2
		# 
		@label2.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label2.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label2.Location = System::Drawing::Point.new(88, 107)
		@label2.Name = "label2"
		@label2.Size = System::Drawing::Size.new(105, 42)
		@label2.TabIndex = 1
		# 
		# label3
		# 
		@label3.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label3.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label3.Location = System::Drawing::Point.new(88, 166)
		@label3.Name = "label3"
		@label3.Size = System::Drawing::Size.new(105, 42)
		@label3.TabIndex = 2
		# 
		# label4
		# 
		@label4.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label4.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label4.Location = System::Drawing::Point.new(88, 230)
		@label4.Name = "label4"
		@label4.Size = System::Drawing::Size.new(105, 42)
		@label4.TabIndex = 3
		# 
		# label5
		# 
		@label5.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label5.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label5.Location = System::Drawing::Point.new(199, 45)
		@label5.Name = "label5"
		@label5.Size = System::Drawing::Size.new(105, 42)
		@label5.TabIndex = 4
		# 
		# label6
		# 
		@label6.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label6.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label6.Location = System::Drawing::Point.new(199, 107)
		@label6.Name = "label6"
		@label6.Size = System::Drawing::Size.new(105, 42)
		@label6.TabIndex = 5
		# 
		# label7
		# 
		@label7.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label7.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label7.Location = System::Drawing::Point.new(199, 166)
		@label7.Name = "label7"
		@label7.Size = System::Drawing::Size.new(105, 42)
		@label7.TabIndex = 6
		# 
		# label8
		# 
		@label8.BackColor = System::Drawing::SystemColors.MenuHighlight
		@label8.ForeColor = System::Drawing::SystemColors.ControlLightLight
		@label8.Location = System::Drawing::Point.new(199, 230)
		@label8.Name = "label8"
		@label8.Size = System::Drawing::Size.new(105, 42)
		@label8.TabIndex = 7
		# 
		# button1
		# 
		@button1.BackColor = System::Drawing::SystemColors.ActiveCaption
		@button1.Location = System::Drawing::Point.new(517, 45)
		@button1.Name = "button1"
		@button1.Size = System::Drawing::Size.new(150, 49)
		@button1.TabIndex = 8
		@button1.Text = "Show"
		@button1.UseVisualStyleBackColor = false
		@button1.Click { |sender, e| self.Button1Click(sender, e) }
		# 
		# button2
		# 
		@button2.Location = System::Drawing::Point.new(517, 146)
		@button2.Name = "button2"
		@button2.Size = System::Drawing::Size.new(150, 42)
		@button2.TabIndex = 9
		@button2.Text = "Hide"
		@button2.UseVisualStyleBackColor = true
		@button2.Click { |sender, e| self.Button2Click(sender, e) }
		# 
		# MainForm
		# 
		self.BackColor = System::Drawing::SystemColors.ActiveCaptionText
		self.ClientSize = System::Drawing::Size.new(829, 319)
		self.Controls.Add(@button2)
		self.Controls.Add(@button1)
		self.Controls.Add(@label8)
		self.Controls.Add(@label7)
		self.Controls.Add(@label6)
		self.Controls.Add(@label5)
		self.Controls.Add(@label4)
		self.Controls.Add(@label3)
		self.Controls.Add(@label2)
		self.Controls.Add(@label1)
		self.Name = "MainForm"
		self.Text = "Schedule"
		self.ResumeLayout(false)
	end

	def Button2Click(sender, e)
		@label1.Text = ""
		@label2.Text = ""
		@label3.Text = ""
		@label4.Text = ""
		@label5.Text = ""
		@label6.Text = ""
		@label7.Text = ""
		@label8.Text = ""
	end

	def Label1Click(sender, e)
		
	end

	def Button1Click(sender, e)
		@label1.Text = "1. Geometry Honors"
		@label2.Text = "2. Band"
		@label3.Text = "3. Computer Programming"
		@label4.Text = "4. English"
		@label5.Text = "5. Gym"
		@label6.Text = "6. Automotive"
		@label7.Text = "7. US History"
		@label8.Text = "8. Chemistry Honors"
	end
end

File: prog54a/MainForm.py
﻿import System.Drawing
import System.Windows.Forms

from System.Drawing import *
from System.Windows.Forms import *

class MainForm(Form):
	def __init__(self):
		self.InitializeComponent()
	
	def InitializeComponent(self):
		self._comboBox1 = System.Windows.Forms.ComboBox()
		self._label1 = System.Windows.Forms.Label()
		self._button1 = System.Windows.Forms.Button()
		self._button2 = System.Windows.Forms.Button()
		self._button3 = System.Windows.Forms.Button()
		self._label2 = System.Windows.Forms.Label()
		self._label3 = System.Windows.Forms.Label()
		self._label4 = System.Windows.Forms.Label()
		self._label5 = System.Windows.Forms.Label()
		self._label6 = System.Windows.Forms.Label()
		self._label7 = System.Windows.Forms.Label()
		self.SuspendLayout()
		# 
		# comboBox1
		# 
		self._comboBox1.Font = System.Drawing.Font("Microsoft Sans Serif", 12, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._comboBox1.FormattingEnabled = True
		self._comboBox1.Items.AddRange(System.Array[System.Object](
			["1970 VW Bug",
			"1979 Firebird",
			"1980 Subaru",
			"1975 Cutlass"]))
		self._comboBox1.Location = System.Drawing.Point(183, 6)
		self._comboBox1.Name = "comboBox1"
		self._comboBox1.Size = System.Drawing.Size(595, 28)
		self._comboBox1.TabIndex = 0
		# 
		# label1
		# 
		self._label1.Font = System.Drawing.Font("Microsoft Sans Serif", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label1.Location = System.Drawing.Point(-3, 1)
		self._label1.Name = "label1"
		self._label1.Size = System.Drawing.Size(166, 37)
		self._label1.TabIndex = 1
		self._label1.Text = "Pick a car:"
		self._label1.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
		self._label1.Click += self.Label1Click
		# 
		# button1
		# 
		self._button1.BackColor = System.Drawing.Color.Chocolate
		self._button1.Font = System.Drawing.Font("Microsoft Sans Serif", 15.75, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._button1.Location = System.Drawing.Point(12, 252)
		self._button1.Name = "button1"
		self._button1.Size = System.Drawing.Size(141, 70)
		self._button1.TabIndex = 2
		self._button1.Text = "Calculate"
		self._button1.UseVisualStyleBackColor = False
		self._button1.Click += self.Button1Click
		# 
		# button2
		# 
		self._button2.BackColor = System.Drawing.Color.Chocolate
		self._button2.Font = System.Drawing.Font("Microsoft Sans Serif", 15.75, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._button2.Location = System.Drawing.Point(310, 252)
		self._button2.Name = "button2"
		self._button2.Size = System.Drawing.Size(141, 70)
		self._button2.TabIndex = 3
		self._button2.Text = "Clear"
		self._button2.UseVisualStyleBackColor = False
		self._button2.Click += self.Button2Click
		# 
		# button3
		# 
		self._button3.BackColor = System.Drawing.Color.Chocolate
		self._button3.Font = System.Drawing.Font("Microsoft Sans Serif", 15.75, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._button3.Location = System.Drawing.Point(646, 252)
		self._button3.Name = "button3"
		self._button3.Size = System.Drawing.Size(141, 70)
		self._button3.TabIndex = 4
		self._button3.Text = "Exit"
		self._button3.UseVisualStyleBackColor = False
		self._button3.Click += self.Button3Click
		# 
		# label2
		# 
		self._label2.BackColor = System.Drawing.Color.Sienna
		self._label2.Font = System.Drawing.Font("Microsoft Sans Serif", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label2.Location = System.Drawing.Point(12, 60)
		self._label2.Name = "label2"
		self._label2.Size = System.Drawing.Size(178, 45)
		self._label2.TabIndex = 5
		self._label2.Text = "Miles:"
		self._label2.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
		# 
		# label3
		# 
		self._label3.BackColor = System.Drawing.Color.Sienna
		self._label3.Font = System.Drawing.Font("Microsoft Sans Serif", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label3.Location = System.Drawing.Point(12, 131)
		self._label3.Name = "label3"
		self._label3.Size = System.Drawing.Size(178, 45)
		self._label3.TabIndex = 6
		self._label3.Text = "Gallons:"
		self._label3.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
		# 
		# label4
		# 
		self._label4.BackColor = System.Drawing.Color.Sienna
		self._label4.Font = System.Drawing.Font("Microsoft Sans Serif", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label4.Location = System.Drawing.Point(12, 195)
		self._label4.Name = "label4"
		self._label4.Size = System.Drawing.Size(178, 45)
		self._label4.TabIndex = 7
		self._label4.Text = "MPG:"
		self._label4.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
		# 
		# label5
		# 
		self._label5.BackColor = System.Drawing.Color.Sienna
		self._label5.Font = System.Drawing.Font("Microsoft Sans Serif", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label5.Location = System.Drawing.Point(273, 195)
		self._label5.Name = "label5"
		self._label5.Size = System.Drawing.Size(505, 45)
		self._label5.TabIndex = 10
		self._label5.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
		# 
		# label6
		# 
		self._label6.BackColor = System.Drawing.Color.Sienna
		self._label6.Font = System.Drawing.Font("Microsoft Sans Serif", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label6.Location = System.Drawing.Point(273, 131)
		self._label6.Name = "label6"
		self._label6.Size = System.Drawing.Size(505, 45)
		self._label6.TabIndex = 9
		self._label6.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
		# 
		# label7
		# 
		self._label7.BackColor = System.Drawing.Color.Sienna
		self._label7.Font = System.Drawing.Font("Microsoft Sans Serif", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label7.Location = System.Drawing.Point(273, 60)
		self._label7.Name = "label7"
		self._label7.Size = System.Drawing.Size(505, 45)
		self._label7.TabIndex = 8
		self._label7.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
		# 
		# MainForm
		# 
		self.BackColor = System.Drawing.Color.Goldenrod
		self.ClientSize = System.Drawing.Size(799, 334)
		self.Controls.Add(self._label5)
		self.Controls.Add(self._label6)
		self.Controls.Add(self._label7)
		self.Controls.Add(self._label4)
		self.Controls.Add(self._label3)
		self.Controls.Add(self._label2)
		self.Controls.Add(self._button3)
		self.Controls.Add(self._button2)
		self.Controls.Add(self._button1)
		self.Controls.Add(self._label1)
		self.Controls.Add(self._comboBox1)
		self.Name = "MainForm"
		self.Text = "prog54a"
		self.Load += self.MainFormLoad
		self.ResumeLayout(False)


	def MainFormLoad(self, sender, e):
		pass

	def Button1Click(self, sender, e):
		miles = 0
		gallons = 0
		mpg = 0
		car = self._comboBox1.Text
		
		if car == "1970 VW Bug":
			miles = 286
			gallons = 9
		
		elif car == "1979 Firebird":
			miles = 412
			gallons = 40
			
		elif car == "1980 Subaru":
			miles = 361
			gallons = 18
			
		elif car == "1975 Cutlass":
			miles = 161
			gallons = 11
		
		mpg = miles / float(gallons)
		mpg = round(mpg, 10)
		
		self._label7.Text = str(miles)
		self._label6.Text = str(gallons)
		self._label5.Text = str(mpg)

	def Button2Click(self, sender, e):
		self._comboBox1.Text = ""
		self._label7.Text = ""
		self._label6.Text = ""
		self._label5.Text = ""
		

	def Button3Click(self, sender, e):
		Application.Exit()

	def Label1Click(self, sender, e):
		pass